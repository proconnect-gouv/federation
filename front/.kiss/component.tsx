import React from 'react';

interface DumbComponentProps {
  className: string;
}

export const DumbComponent: React.FC<DumbComponentProps> = React.memo(
  ({ className }: DumbComponentProps) => {
    return <div />;
  },
);

DumbComponent.displayName = 'DumbComponent';
