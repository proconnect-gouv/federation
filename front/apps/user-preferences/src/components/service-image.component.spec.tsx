import { render } from '@testing-library/react';

import { ServiceImageComponent } from './service-image.component';

describe('ServiceImageComponent', () => {
  // Given
  const serviceMock = {
    active: true,
    image: undefined,
    isChecked: false,
    name: 'name-mock',
    title: 'title-mock',
    uid: 'uid-mock',
  };

  it('should match the snapshot', () => {
    // When
    const { container } = render(<ServiceImageComponent service={serviceMock} />);

    // Then
    expect(container).toMatchSnapshot();
  });

  it('should show a placeholder instead of an image', () => {
    // When
    const { container, getByText } = render(<ServiceImageComponent service={serviceMock} />);
    const element = getByText('title-mock');

    // Then
    expect(container).toMatchSnapshot();
    expect(element).toBeInTheDocument();
    expect(element.tagName).toBe('B');
  });

  it('should show an image', () => {
    // Given
    const servicesMockWithImage = { ...serviceMock, image: 'image.mock' };

    // When
    const { container, getByAltText } = render(
      <ServiceImageComponent service={servicesMockWithImage} />,
    );
    const element = getByAltText(`fournisseur d'identitÃ© title-mock`);

    // Then
    expect(container).toMatchSnapshot();
    expect(element).toBeInTheDocument();
    expect(element.tagName).toBe('IMG');
  });

  it('should be in a disabled state', () => {
    // Given
    const servicesMockWithImage = { ...serviceMock, image: 'image.mock' };

    // When
    const { container } = render(
      <ServiceImageComponent disabled service={servicesMockWithImage} />,
    );
    const wrapper = container.firstChild;

    // Then
    expect(container).toMatchSnapshot();
    expect(wrapper).toHaveClass('disabled');
    expect(wrapper).toHaveClass('opacity-45');
  });
});
