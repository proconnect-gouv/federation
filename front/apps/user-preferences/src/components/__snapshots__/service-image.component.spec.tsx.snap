// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceImageComponent should be in a disabled state 1`] = `
<div>
  <div
    class="image is-relative no-overflow fr-mr-3w disabled opacity-45"
  >
    <img
      alt="fournisseur d'identité title-mock"
      class="is-block is-absolute"
      height="auto"
      src="/images/fi/image.mock"
      width="auto"
    />
  </div>
</div>
`;

exports[`ServiceImageComponent should match the snapshot 1`] = `
<div>
  <div
    class="image is-relative no-overflow fr-mr-3w is-table"
  >
    <b
      class="is-table-cell v-align-middle text-center w100 fr-text--sm"
    >
      title-mock
    </b>
  </div>
</div>
`;

exports[`ServiceImageComponent should show a placeholder instead of an image 1`] = `
<div>
  <div
    class="image is-relative no-overflow fr-mr-3w is-table"
  >
    <b
      class="is-table-cell v-align-middle text-center w100 fr-text--sm"
    >
      title-mock
    </b>
  </div>
</div>
`;

exports[`ServiceImageComponent should show an image 1`] = `
<div>
  <div
    class="image is-relative no-overflow fr-mr-3w"
  >
    <img
      alt="fournisseur d'identité title-mock"
      class="is-block is-absolute"
      height="auto"
      src="/images/fi/image.mock"
      width="auto"
    />
  </div>
</div>
`;
